src/test/java/spoon/test/ctCase/CtCaseImplTest.java

```diff
@@ -0,0 +1,58 @@
+package spoon.test.ctCase;
+
+import org.junit.jupiter.api.Test;
+import spoon.Launcher;
+import spoon.reflect.code.CtCase;
+import spoon.reflect.code.CtStatement;
+import spoon.reflect.code.CtStatementList;
+import spoon.reflect.declaration.CtClass;
+import spoon.reflect.visitor.filter.TypeFilter;
+import spoon.support.compiler.FileSystemFile;
+import spoon.support.reflect.code.CtStatementListImpl;
+import spoon.test.ctCase.testclasses.ClassWithSwitchExample;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class CtCaseImplTest {
+
+    @Test
+    void testInsertBeginWithListOfStatements() {
+        // contract: insertBegin adds a list of statements at the beginning of a case, i.e the list is added above two
+        // exiting statements, and the order of statements in the list remains the same
+
+        // arrange
+        CtClass classWithSwitchExample = getClassWithSwitchExample();
+        CtStatement firstStatementToBeInserted = createStatement(classWithSwitchExample, "firstStatement");
+        CtStatement secondStatementToBeInserted = createStatement(classWithSwitchExample, "secondStatement");
+        CtCase testCase = classWithSwitchExample.getElements(new TypeFilter<>(CtCase.class)).get(0);
+
+        CtStatementList ctStatementList =  new CtStatementListImpl<CtStatement>() {
+            {
+                addStatement(firstStatementToBeInserted);
+                addStatement(secondStatementToBeInserted);
+            }
+        };
+
+        // act
+        testCase.insertBegin(ctStatementList);
+
+        // assert
+        int firstStatementIndex = 0;
+        int secondStatementIndex = 1;
```

Comment:
Perhaps a bit overkill naming these indices :)

Reply:
yes it seemed to me as well 😅